runelite-client/src/main/java/net/runelite/client/plugins/slayer/SlayerPlugin.java

```diff
@@ -130,9 +134,13 @@ public void onGameStateChange(GameStateChanged event)
 				amount = 0;
 				break;
 			case LOGGED_IN:
-				if (config.amount() != -1 && !config.taskName().isEmpty())
+				if (!client.getUsername().equals(lastUsername))
```

Comment:
What is the reason for this logic?

Reply:
it prevents setTask being called for every region change (timer setting is done within setTask)

Reply:
Wouldn't this also stop it from being called if you just logged out and in? It might be better to set a flag for LOGGING_IN and then check it in LOGGED_IN. I think some of the other plugins do this.

Reply:
changed as requested